### PR TITLE
Add support for ASRock Z790 Taichi/Z790 Taichi Carrara.

### PR DESCRIPTION
**Pull request by papa12804**
Add support for fans and temps, but voltage readings in NCT6686D chip#1 is range from 0x106 to 0x114.
The `_voltageRegisters` in `Nct677X` is range from 0x120 to 0x13C, there needs some revise.

### CODE REVIEW

**Comment by PhyxionNL** [comment]
Do you still want to work on this PR @papa12804?

**Comment by papa12804** [comment]
I can't come up with some good ideas for this now, adding brand judgments to `Nct677X` will make it messy.

**Comment by PhyxionNL** [comment]
Hmm, NCT6686D (and NCT6683D) should use the same voltage registers as NCT6687D. I'll leave some feedback though :)

File: LibreHardwareMonitorLib/Hardware/Motherboard/Identification.cs [hunk 1]
```diff
@@ -369,6 +369,10 @@ public static Model GetModel(string name)
                 return Model.ROG_MAXIMUS_Z790_HERO;
             case var _ when name.Equals("PRIME Z690-A", StringComparison.OrdinalIgnoreCase):
                 return Model.PRIME_Z690_A;
+            case var _ when name.Equals("Z790 Taichi", StringComparison.OrdinalIgnoreCase):
+                return Model.Z790_Taichi;
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Motherboard/Identification.cs:373]: This line can go away.

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs [hunk 2]
```diff
@@ -355,47 +355,91 @@ private static void GetBoardSpecificConfiguration
...
+                                // v.Add(new Voltage("VCCSA Voltage", 5));
+                                // v.Add(new Voltage("VCCIN_AUX Voltage", 6));
+                                // v.Add(new Voltage("VDD_CPU Voltage", 7));
+                                // v.Add(new Voltage("Voltage #8", 8, true));
+                                // v.Add(new Voltage("Voltage #9", 9, true));
+                                // v.Add(new Voltage("Voltage #10", 10, true));
+                                // v.Add(new Voltage("Voltage #11", 11, true));
+                                // v.Add(new Voltage("Voltage #12", 12, true));
+                                // v.Add(new Voltage("Voltage #13", 13, true));
+
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:378]: Remove

File: LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs [hunk 3]
```diff
@@ -2289,6 +2333,22 @@ private static void GetNuvotonConfigurationD(ISuperIO superIO, Manufacturer manu
                         c.Add(new Ctrl("Chipset Fan", 3));
                         break;
 
+                    case Model.Z790_Taichi:
+                        v.Add(new Voltage("+1.8V PROC Voltage", 0));
```
Comment (PhyxionNL) [inline LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs:2337]: Follow file naming scheme:
PROC > CPU
PCH > Chipset.

Also the + can be removed from 1.8, 0.82 and 1.05V (but keep 3.3V).

**Review by PhyxionNL — CHANGES_REQUESTED** [review]

**Comment by papa12804** [comment]
> Hmm, NCT6686D (and NCT6683D) should use the same voltage registers as NCT6687D. I'll leave some feedback though :)

I think each brand has its own NCT6686D or NCT6683D or NCT6687D firmware, so the location of the fan, temperature and voltage registers may vary.

**Pull request merged**